client/rpc/src/lib.rs

```diff
@@ -331,8 +330,26 @@ where
         // Check if the execution is safe.
         check_if_safe(ext)?;
 
-        // Open file in the local file system.
-        let mut file = File::open(PathBuf::from(file_path.clone())).map_err(into_rpc_error)?;
+        // Create file handler for local or remote file.
+        let config = RemoteFileConfig::new(DEFAULT_MAX_FILE_SIZE);
+        let (handler, url) = RemoteFileHandlerFactory::create_from_string(&file_path, config)
+            .map_err(|e| into_rpc_error(format!("Failed to create file handler: {:?}", e)))?;
+
+        let file_size = handler
+            .get_file_size()
+            .await
+            .map_err(remote_file_error_to_rpc_error)?;
+
+        // For local files, check if the file is empty
+        // Remote files we allow the server to give us "0" as file size to support dynamic content
+        if (url.scheme() == "" || url.scheme() == "file") && file_size == 0 {
+            return Err(into_rpc_error(FileStorageError::FileIsEmpty));
+        }
+
+        let mut stream = handler
+            .stream_file()
+            .await
+            .map_err(remote_file_error_to_rpc_error)?;
 
         // Instantiate an "empty" [`FileDataTrie`] so we can write the file chunks into it.
         let mut file_data_trie = self.file_storage.write().await.new_file_data_trie();
```

Comment:
I also noticed that we acquire a `.write()` lock here where a `.read()` one would suffice, since we are creating a new trie. This ties a bit with the file locks we discussed in another thread. 

But yeah until we want to add the file trie to the file storage we don't need to hold locks